egs/avspeech/local/loader/data.py

```diff
@@ -0,0 +1,125 @@
+import os
+import cv2
+import librosa
+import numpy as np
+from pathlib import Path
+
+from typing import Callable, Tuple, List
+
+def get_frames(video):
+    frame_count = int(video.get(cv2.CAP_PROP_FRAME_COUNT))
+    frame_width = int(video.get(cv2.CAP_PROP_FRAME_WIDTH))
+    frame_height = int(video.get(cv2.CAP_PROP_FRAME_HEIGHT))
+
+    buffer_video = np.empty((frame_count, frame_height, frame_width, 3), np.dtype('uint8'))
+
+    frame = 0
+    ret = True
+
+    while (frame < frame_count  and ret):
+        ret, f = video.read()
+        buffer_video[frame] = cv2.cvtColor(f, cv2.COLOR_BGR2RGB)
+
+        frame += 1
+    video.release()
+    return buffer_video
+
+class Signal:
```

Comment:
Could you refactor this to be in the same file as the `AVSpeechDataset`? I don't see the point on passing the to the dataset? Is it meant to be changed or not?

Reply:
The whole file, right? Yeah, I guess it would be better if we just move it into `avspeech_dataset.py`. However `generate_video_embedding.py` needs `get_frames` as well. So, we can move that function to `asteroid.utils` and import it as `from asteroid.utils import get_frames`?

Reply:
Because this is the only dataset that uses video stuff for now, I'd put all the related functions into `avspeech_dataset.py` and add optional requirements in the setup. The install is already pretty heavy and avspeech will need lots of stuff. Does it sound ok for you?